File: app/src/main/java/com/example/files/utils/Animations.java
package com.example.files.utils;

import static com.example.files.Statics.dpToPixels;

import android.animation.Animator;
import android.view.View;
import android.view.animation.AccelerateInterpolator;
import android.view.animation.DecelerateInterpolator;

public class Animations {

    public static void show(View actionBar, Runnable onAnimationEnd) {
        actionBar.animate()
                .translationY(0)
                .setInterpolator(new DecelerateInterpolator(2))
                .setListener(new Animator.AnimatorListener() {
                    @Override
                    public void onAnimationStart(Animator animation) {
                        actionBar.setVisibility(View.VISIBLE);
                    }

                    @Override
                    public void onAnimationEnd(Animator animation) {
                        onAnimationEnd.run();
                    }

                    @Override
                    public void onAnimationCancel(Animator animation) {

                    }

                    @Override
                    public void onAnimationRepeat(Animator animation) {

                    }
                }).start();
    }


    public static void hide(View actionBar) {
        actionBar.animate()
                .translationY(dpToPixels(actionBar.getHeight() + 10))
                .setInterpolator(new AccelerateInterpolator(2))
                .setListener(endListener(() -> actionBar.setVisibility(View.GONE))).start();
    }

    public static void hide(View actionBar, Runnable onAnimationEnd) {
        actionBar.animate()
                .translationY(dpToPixels(actionBar.getHeight() + 10))
                .setInterpolator(new AccelerateInterpolator(2))
                .setListener(endListener(onAnimationEnd)).start();
    }

    public static Animator.AnimatorListener endListener(Runnable onAnimationEnd) {
        return new Animator.AnimatorListener() {
            @Override
            public void onAnimationStart(Animator animation) {
            }

            @Override
            public void onAnimationEnd(Animator animation) {
                onAnimationEnd.run();
            }

            @Override
            public void onAnimationCancel(Animator animation) {

            }

            @Override
            public void onAnimationRepeat(Animator animation) {

            }
        };
    }
}

File: app/src/main/java/com/example/files/utils/Animations.kt
package com.example.files.utils

import android.animation.Animator
import android.view.View
import android.view.animation.AccelerateInterpolator
import android.view.animation.DecelerateInterpolator
import com.example.files.Statics.dpToPixels

object Animations {

    @JvmStatic
    fun show(actionBar: View, onAnimationEnd: Runnable) {
        actionBar.animate()
            .translationY(0f)
            .setInterpolator(DecelerateInterpolator(2f))
            .setListener(object : Animator.AnimatorListener {
                override fun onAnimationStart(animation: Animator) {
                    actionBar.visibility = View.VISIBLE
                }

                override fun onAnimationEnd(animation: Animator) {
                    onAnimationEnd.run()
                }

                override fun onAnimationCancel(animation: Animator) {}
                override fun onAnimationRepeat(animation: Animator) {}
            }).start()
    }

    @JvmStatic
    fun hide(actionBar: View) {
        actionBar.animate()
            .translationY(dpToPixels((actionBar.height + 10).toFloat()).toFloat())
            .setInterpolator(AccelerateInterpolator(2f))
            .setListener(endListener { actionBar.visibility = View.GONE }).start()
    }

    @JvmStatic
    fun hide(actionBar: View, onAnimationEnd: Runnable) {
        actionBar.animate()
            .translationY(dpToPixels((actionBar.height + 10).toFloat()).toFloat())
            .setInterpolator(AccelerateInterpolator(2f))
            .setListener(endListener(onAnimationEnd)).start()
    }

    @JvmStatic
    fun endListener(onAnimationEnd: Runnable): Animator.AnimatorListener {
        return object : Animator.AnimatorListener {
            override fun onAnimationStart(animation: Animator) {}
            override fun onAnimationEnd(animation: Animator) {
                onAnimationEnd.run()
            }

            override fun onAnimationCancel(animation: Animator) {}
            override fun onAnimationRepeat(animation: Animator) {}
        }
    }
}
